cmd/ctl/pkg/install/helm/resource.go

```diff
@@ -0,0 +1,87 @@
+/*
+Copyright 2021 The cert-manager Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package helm
+
+import (
+	"bytes"
+	"fmt"
+
+	"k8s.io/apimachinery/pkg/api/meta"
+	"k8s.io/cli-runtime/pkg/resource"
+
+	"helm.sh/helm/v3/pkg/kube"
+)
+
+const (
+	customResourceDefinitionGroup = "apiextensions.k8s.io"
+	customResourceDefinitionKind  = "CustomResourceDefinition"
+)
+
+// Build a list of resource.Info objects from a rendered manifest.
+func GetChartResourceInfo(manifest string, kubeClient kube.Interface) ([]*resource.Info, error) {
+	resources := make([]*resource.Info, 0)
+
+	res, err := kubeClient.Build(bytes.NewBufferString(manifest), false)
+	if err != nil {
+		return nil, fmt.Errorf("failed to parse CRDs from render: %s", err)
+	}
+	resources = append(resources, res...)
+
+	return resources, nil
+}
+
+func filterResources(resources []*resource.Info, filter func(*resource.Info) bool) []*resource.Info {
+	crds := make([]*resource.Info, 0)
+	for _, res := range resources {
+		if filter(res) {
+			crds = append(crds, res)
+		}
+	}
+
+	return crds
+}
```

Comment:
The name of this function doesn't match its content. It appears to be a general filter function with the ability to inject any sort of filter, but it assigns the results to a variable called `crds`.